web-app-router/scenes/Portal/Teams/TeamId/Team/page/Members/List/index.tsx

```diff
@@ -0,0 +1,396 @@
+"use client";
+
+import { useParams } from "next/navigation";
+import {
+  FetchMembershipsQuery,
+  useFetchMembershipsQuery,
+} from "./graphql/client/fetch-members.generated";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import { TYPOGRAPHY, Typography } from "@/components/Typography";
+import { UserLogo } from "./UserLogo";
+import { Role_Enum } from "@/graphql/graphql";
+import {
+  Dropdown,
+  DropdownButton,
+  DropdownItem,
+  DropdownItems,
+} from "@/components/Dropdown";
+import { MoreVerticalIcon } from "@/components/Icons/MoreVerticalIcon";
+import { Footer } from "@/components/Table/Footer";
+import { EditUserIcon } from "@/components/Icons/EditUserIcon";
+import { TrashIcon } from "@/components/Icons/TrashIcon";
+import { useUser } from "@auth0/nextjs-auth0/client";
+import { Auth0SessionUser } from "@/lib/types";
+import { useFetchUserLazyQuery } from "./graphql/client/fetch-user.generated";
+import {
+  FetchInvitesDocument,
+  useFetchInvitesQuery,
+} from "./graphql/client/fetch-invites.generated";
+import dayjs from "dayjs";
+import Skeleton from "react-loading-skeleton";
+import { useAtom } from "jotai";
+import { RemoveUserDialog, removeUserDialogAtom } from "./RemoveUserDialog";
+import { EditRoleDialog, editRoleDialogAtom } from "./EditRoleDialog";
+import { PermissionsDialog } from "./PermissionsDialog";
+import { useInviteTeamMembersMutation } from "../graphql/client/invite-team-members.generated";
+import { toast } from "react-toastify";
+import { SendIcon } from "@/components/Icons/SendIcon";
+import { useDeleteInviteMutation } from "./graphql/client/delete-invite.generated";
+
+const roleName: Record<Role_Enum, string> = {
+  [Role_Enum.Admin]: "Admin",
+  [Role_Enum.Member]: "Member",
+  [Role_Enum.Owner]: "Owner",
+};
+
+export const List = (props: { search?: string }) => {
+  const { user } = useUser() as Auth0SessionUser;
+  const { teamId } = useParams() as { teamId: string };
+  const [, setIsRemoveDialogOpened] = useAtom(removeUserDialogAtom);
+  const [, setIsEditRoleDialogOpened] = useAtom(editRoleDialogAtom);
+
+  const [userToRemove, setUserToRemove] = useState<
+    FetchMembershipsQuery["membership"][number]["user"] | null
+  >(null);
+
+  const [userToEditRole, setUserToEditRole] = useState<
+    FetchMembershipsQuery["membership"][number] | null
+  >(null);
+
+  const [fetchUser, { data: fetchUserResult }] = useFetchUserLazyQuery({
+    context: { headers: { team_id: teamId } },
+  });
+
+  useEffect(() => {
+    if (!user?.hasura.id) {
+      return;
+    }
+
+    fetchUser({
+      variables: {
+        userId: user?.hasura.id,
+        teamId,
+      },
+    });
+  }, [fetchUser, teamId, user?.hasura.id]);
+
+  const isEnoughPermissions = useMemo(() => {
```

Comment:
we can merge this for now but leave a TODO to use the new `isenoughpermissions`. I made a helper so we can standardize logic changes